### DISCUSSION TITLE
Why Was Contiki Forked?

### DISCUSSION THREAD

**Discussion by SaulMoonves** [contiki-ng/contiki-ng#1728] [General]
I'm not complaining, I just wondered what the impetus for splitting off to a new project was.

**Comment by alexrayne** [comment]
they 
- won"t support 8bit architectures, ipv4, cxmac
- dramaticaly reorganise source tree; 
- drop off applications, insteard of it use services, and cleanup code and examples from trash
- delployes some tools and libs to standalone repos
- drop huge oooold useless history, and about of 0.5Gb deleted files in it
and it not all

**Comment by atiselsts** [comment]
We have a Contiki-NG paper under review. The paper aims to provide the official answer to this. I hope George and other authors would not mind if I share it here.

### _From Contiki to Contiki-NG_

_As the Contiki OS gained traction and started being used extensively, some limitations began to emerge_

* _Large legacy of old, extremely resource-constrained platforms. The 8-bit and 16-bit low-power microcontrollers which Contiki was originally designed for became obsolete over time; 32-bit ARM Cortex-M based devices with more sophisticated low-power modes are the new norm._
* _Support for non-standard protocols. As the field of wireless sensor network research evolved to become one of the core enabling technologies of the IoT, interoperability and standards became increasingly important. Alongside standards-compliant protocol implementations, the Contiki code-base also featured older, experimental, non-standard protocols initially contributed as research artefacts. One such example is the Rime stack._

_This combination of legacy platform code and support for non-standard networking protocols increased maintenance complexity and impeded code evolution._

_Contiki-NG was first released in November 2017. Its aim was to eliminate some of Contiki's limitations in order to enable easier maintenance and quicker evolution. The main vision is to focus on:_
* _Standard protocols. Some of the standards supported by Contiki-NG are IEEE 802.15.4 TSCH, 6LoWPAN, 6TiSCH, RPL, CoAP, MQTT, and LWM2M._
* _Support for modern hardware platforms._
* _Dependability (reliability and security) through modern development practices, continuous integration using simulations and a physical testbed, and security testing techniques._

_The first release (v4.0) was a fork of the Contiki OS. Besides the aforementioned changes in the high-level focus of the OS, Contiki-NG added a new configuration and logging system, a new lightweight and reliable RPL implementation (RPL-Lite), and a network administration shell. It also brought with it an extensive cleanup of the codebase, with legacy platforms, protocols and services removed, so as to eliminate constraints on future developments._
